scionlab/views/user_as_views.py

```diff
@@ -294,32 +127,19 @@ class UserASGetConfigView(OwnedUserASQuerysetMixin, SingleObjectMixin, View):
     def get(self, request, *args, **kwargs):
         user_as = self.get_object()
         filename = 'scion_lab_{user}_{ia}.tar.gz'.format(
-                        user=user_as.owner.email,
-                        ia=user_as.isd_as_path_str())
+            user=user_as.owner.email,
+            ia=user_as.isd_as_path_str())
         resp = HttpResponseAttachment(filename=filename, content_type='application/gzip')
         with closing(tarfile.open(mode='w:gz', fileobj=resp)) as tar:
             config_tar.generate_user_as_config_tar(user_as, TarWriter(tar))
         return resp
 
 
+# TODO(andrea_tulimiero): We can use the OwnedUserASQuerysetMixin here
```

Comment:
I see, and I think we can solve by defining `OwnedUserASQuerysetMixin` like this:
```
class OwnedUserASQuerysetMixin:
    """
    Defines get_queryset to get only the UserASes owned by
    the current user.
    To be used in a View that uses `django.views.generic.detail.SingleObjectMixin`
    """

    def get_queryset(self):
        return super().get_queryset().filter(owner=self.request.user)
```
This way it uses the ordering defined by the parent, hence respecting the `ordering = ['as_id']`

Reply:
If i recall correctly, this does not work because the single object views don't have a `get_queryset` function. But if you can make it work, go for it.